### PR TITLE
secure_demo: generate statically-keyed demo image

### PR DESCRIPTION
**Pull request by gtank**
This format builds a disk with a single ESP partition containing
lockdown.efi, a signed shim, and a CoreOS image configured to boot
using UEFI secure boot.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "build_library/secure_demo/shim"]
```
Comment (marineam) [inline .gitmodules]: Please use the repo manifest instead. It is going to be confusing to use both git submodules and repo.

File: build_library/secure_demo/find_gnuefi.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+diff --git a/Makefile b/Makefile
+index 332a29b..b8359e7 100644
+--- a/Makefile
++++ b/Makefile
+@@ -10,7 +10,7 @@ LIB_PATH	= /usr/lib64
+ 
+ EFI_INCLUDE	:= /usr/include/efi
+ EFI_INCLUDES	= -nostdinc -ICryptlib -ICryptlib/Include -I$(EFI_INCLUDE) -I$(EFI_INCLUDE)/$(ARCH) -I$(EFI_INCLUDE)/protocol -Iinclude
+-EFI_PATH	:= /usr/lib64/gnuefi
++EFI_PATH	:= $(shell find /usr/lib* -name "crt0-efi-x86_64.o" -exec dirname {} \;)
```
Comment (marineam) [inline build_library/secure_demo/find_gnuefi.patch]: Alternatively, could we just pass EFI_PATH=/usr/libwhatever to make instead?

File: build_library/vm_image_util.sh [hunk 3]
```diff
@@ -914,36 +914,132 @@ _write_secure_demo_disk() {
     local dst_img="$2"
     local tmp_esp="${VM_TMP_DIR}/esp"
 
+    local workdir=`pwd`
```
Comment (marineam) [inline build_library/vm_image_util.sh]: looks left over from your older script. please delete

File: build_library/vm_image_util.sh [hunk 4]
```diff
@@ -914,36 +914,132 @@ _write_secure_demo_disk() {
...
+    local root=CoreOS-Boot-CA
+    local signer=CoreOS-Boot-Signer
+    local duration=3650
+
+  ## Format disk image
+
+    echo "Formatting ${dst_img}"
+    sudo losetup -a
```
Comment (marineam) [inline build_library/vm_image_util.sh]: left over debuging?

File: build_library/vm_image_util.sh [hunk 5]
```diff
@@ -914,36 +914,132 @@ _write_secure_demo_disk() {
...
+    sudo cp shim.signed.efi "${tmp_esp}/EFI/boot/bootx64.efi"
+
+    cd ${BUILD_LIBRARY_DIR}/secure_demo
+
+  ## GRUB
+
+    # try to speed things up a bit
+    if [ -e /usr/sbin/rngd ]; then
+      echo using rngd with /dev/urandom
+      sudo rngd -r /dev/urandom
```
Comment (marineam) [inline build_library/vm_image_util.sh]: Please do not start system services in a build script. Do that on the host machine in the usual way...

Comment (marineam) [inline build_library/vm_image_util.sh]: Also, feeding rngd with urandom? seriously? ಠ_ಠ

**Comment by gtank** [comment]
this is now both obsolete and irrelevant

**Pull request closed without merging**